### PR TITLE
Go proof

### LINKED ISSUE #154: Provide a status for each example in the README
In the README file, you can found a dozen configurations and usage examples. In order to double-check that those examples are valid and actually works, I suggest creating a GitHub action and provide a status for each example.  

Progress:
 - [x] - Binary
 - [x] - Maven
 - [ ] - RPM
 - [x] - Docker
 - [x] - Helm
 - [x] - NPM
 - [ ] - Composer
 - [ ] - NuGet
 - [x] - Gem
 - [x] - Pypi
 - [ ] - Go

### LINKED ISSUE #540: Go adapter is broken
When trying to use a go registry:
```txt
$ GO111MODULE=on GOPROXY=http://localhost:8080/my-go-proxy go get -x -insecure golang.org/x/time
# get http://localhost:8080/my-go-proxy/golang.org/@v/list
# get http://localhost:8080/my-go-proxy/golang.org/x/time/@v/list
# get http://localhost:8080/my-go-proxy/golang.org/x/@v/list
# get http://localhost:8080/my-go-proxy/golang.org/x/@v/list: 500 Internal Server Error (0.601s)
# get http://localhost:8080/my-go-proxy/golang.org/x/time/@v/list: 500 Internal Server Error (0.602s)
# get http://localhost:8080/my-go-proxy/golang.org/@v/list: 500 Internal Server Error (0.602s)
go get golang.org/x/time: module golang.org/x/time: reading http://localhost:8080/my-go-proxy/golang.org/x/time/@v/list: 500 Internal Server Error
```

### PR DESCRIPTION
**Pull request by Sammers21**
Closes #154
Closes #540

### CODE REVIEW

File: examples/go/run.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+# Start artipie.
+docker run --rm -d --name artipie -it -v $(pwd)/artipie.yaml:/etc/artipie.yml -v $(pwd):/var/artipie -p 8080:80 artipie/artipie:latest
+
+# Wait for container to be ready for new connections.
+sleep 5
+
+# Force go client to use Aritpie Go registry.
+export GO111MODULE=on
+export GOPROXY=http://localhost:8080/my-go-proxy
```
Comment (olegmoz) [inline examples/go/run.sh:12]: @Sammers21 I see that Go uses word proxy for custom artifacts manager, but it is a bit misleading in context of Artipie. We have proxy type of adapters and someday going to have proxy for Go as well. I'd suggest to remove proxy word from repo name

**Review by olegmoz — CHANGES_REQUESTED** [review]
@Sammers21 thanks, all looks good, have one concern though. Please see the comment

**Review by olegmoz — APPROVED** [review]
@Sammers21 thanks, merging

**Review by olegmoz — CHANGES_REQUESTED** [review]
@Sammers21 the proof is failing after recent changes, please fix

**Review by olegmoz — APPROVED** [review]
@Sammers21 thanks

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#541` is not assigned, can't get performer

**Pull request merged**